packages/manager/modules/pci/src/projects/project/instances/instances.service.js

```diff
@@ -162,31 +163,37 @@ export default class PciProjectInstanceService {
           ipReverse: this.getReverseIp(instance),
         }),
       )
-      .then(
-        ({ instance, ipReverse, volumes, privateNetworks }) =>
-          new Instance({
-            ...instance,
-            flavor: {
-              ...instance.flavor,
-              capabilities: this.constructor.transformCapabilities(
-                get(instance.flavor, 'capabilities', []),
-              ),
-            },
-            volumes: filter(volumes, (volume) =>
-              includes(volume.attachedTo, instance.id),
+      .then(({ instance, ipReverse, volumes, privateNetworks }) => {
+        const localZones = customerRegions?.filter(({ type }) =>
+          type.includes(LOCAL_ZONE_REGION),
+        );
+        const isLocalZone = localZones?.some(
+          (region) => region.name === instance.region,
+        );
+        return new Instance({
+          ...instance,
+          flavor: {
+            ...instance.flavor,
+            capabilities: this.constructor.transformCapabilities(
+              get(instance.flavor, 'capabilities', []),
             ),
-            privateNetworks: filter(privateNetworks, (privateNetwork) =>
-              includes(
-                map(
-                  filter(instance.ipAddresses, { type: 'private' }),
-                  'networkId',
-                ),
-                privateNetwork.id,
+          },
+          volumes: filter(volumes, (volume) =>
+            includes(volume.attachedTo, instance.id),
+          ),
+          privateNetworks: filter(privateNetworks, (privateNetwork) =>
```

Comment:
```suggestion
          privateNetworks: privateNetworks.filter((privateNetwork) =>
              instance.ipAddresses.filter((address) => address.type === 'private').map((instance) => instance.networkId).includes(privateNetwork.id)
```
Not 100% sure of the suggestion as it's hard to read / understand the code